From 2f0564d64789182923db8360fe69f1ed5f262929 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Wed, 20 Dec 2017 15:13:33 +0200
Subject: [PATCH 3131/5242] MLK-17274: drm: imx: dcss: use_global_alpha not
 working if channel already enabled

commit  b09c58a4decdbaa8cd8f3ad7f572ae22a3c62b8c from
https://source.codeaurora.org/external/imx/linux-imx.git

If the channel is already enabled, or it doesn't need a mode set, then
the dcss_dtg_global_alpha_changed() will always return false for formats
with per-pixel alpha. Hence, the plane will not be updated. This patch
removes the check for image format and the check will be done for all
image formats.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-dtg.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-dtg.c b/drivers/gpu/imx/dcss/dcss-dtg.c
index b3b9fc8..cf87df3 100644
--- a/drivers/gpu/imx/dcss/dcss-dtg.c
+++ b/drivers/gpu/imx/dcss/dcss-dtg.c
@@ -309,8 +309,7 @@ bool dcss_dtg_global_alpha_changed(struct dcss_soc *dcss, int ch_num,
 	if (ch_num)
 		return false;
 
-	return dcss_dtg_global_alpha_needed(pix_format) &&
-	       (alpha != dtg->alpha || use_global_alpha != dtg->use_global);
+	return alpha != dtg->alpha || use_global_alpha != dtg->use_global;
 }
 EXPORT_SYMBOL(dcss_dtg_global_alpha_changed);
 
-- 
1.7.9.5

